src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral2.java

```diff
@@ -63,4 +63,36 @@ void method() throws Exception {
             a.equals("");
         }
     }
+
```

Comment:
Please add test cases similar to https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/3fe931a_2022125544/reports/diff/openjdk17/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/openjdk17/test/hotspot/jtreg/vmTestbase/jit/t/t091/t091.java.html#L58 (a bunch of consecutive `if` statements without block), where we have acceptable usage and violations. Also, I am having a hard time figuring out which `if` statement we missed previously; the one immediately preceding the `if` statement where we use the variable?

Reply:
> Please add test cases similar to........

Done.

>Also, I am having a hard time figuring out which if statement we missed previously; the one immediately preceding the if statement where we use the variable?

I will make it clear which `if` statements we missed-

Previous Logic:
```java
private static DetailAST getFirstNodeInsideIfBlock(
        DetailAST block, DetailAST variable) {
    DetailAST firstNodeInsideBlock = null;


    if (!isVariableInOperatorExpr(block, variable)) {
        DetailAST currentNode = block.getLastChild();
        final List<DetailAST> variableUsageExpressions =
                new ArrayList<>();


        while (currentNode != null
                && currentNode.getType() == TokenTypes.LITERAL_ELSE) {
            final DetailAST previousNode =
                    currentNode.getPreviousSibling();


            // Checking variable usage inside IF block.
            if (isChild(previousNode, variable)) {
                variableUsageExpressions.add(previousNode);
            }


            // Looking into ELSE block, get its first child and analyze it.
            currentNode = currentNode.getFirstChild();


            if (currentNode.getType() == TokenTypes.LITERAL_IF) {
                currentNode = currentNode.getLastChild();
            }
            else if (isChild(currentNode, variable)) {
                variableUsageExpressions.add(currentNode);
                currentNode = null;
            }
        }


        // If IF block doesn't include ELSE then analyze variable usage
        // only inside IF block.
        if (currentNode != null
                && isChild(currentNode, variable)) {
            variableUsageExpressions.add(currentNode);
        }


        // If variable usage exists in several related blocks, then
        // firstNodeInsideBlock = null, otherwise if variable usage exists
        // only inside one block, then get node from
        // variableUsageExpressions.
        if (variableUsageExpressions.size() == 1) {
            firstNodeInsideBlock = variableUsageExpressions.get(0);
        }
    }


    return firstNodeInsideBlock;
}
```
If we see the initial logic, we get the if statements last-child, for if statements without `{}`, it will be:
```java
if (true)    
    method();
```
AST-
```
LITERAL_IF -> if 
   |--LPAREN -> ( 
   |--EXPR -> EXPR 
   |   `--LITERAL_TRUE -> true 
   |--RPAREN -> ) 
   |--EXPR -> EXPR 
   |   `--METHOD_CALL -> ( 
   |       |--IDENT -> method 
   |       |--ELIST -> ELIST 
   |       `--RPAREN -> ) 
   `--SEMI -> ; 
```
Now the last child is `SEMI` and the according to the logic, it was supposed to be if statements body.
The logic was built for if statements with `{}` where the last child is `SLIST` i.e. if statement's body.
```
LITERAL_IF -> if 
  |--LPAREN -> ( 
  |--EXPR -> EXPR 
  |   `--LITERAL_TRUE -> true 
  |--RPAREN -> ) 
  `--SLIST -> { 
      |--EXPR -> EXPR 
      |   `--METHOD_CALL -> ( 
      |       |--IDENT -> method 
      |       |--ELIST -> ELIST 
      |       `--RPAREN -> ) 
      |--SEMI -> ; 
      `--RCURLY -> } 
```
Cases like this in combination with `else` were missing, support for that was added.

Reply:
Please add this analysis to the PR description to explain change in behavior of check.

Reply:
Done.